signal.c

```diff
@@ -107,8 +113,48 @@ static const struct eventop evsigops = {
 /* Lock for evsig_base and evsig_base_n_signals_added fields. */
 static void *evsig_base_lock = NULL;
 #endif
-/* The event base that's currently getting informed about signals. */
+
+#if defined(_MSC_VER)
+// Microsoft explicitly excludes stdatomics.h from the list of supported
+// C11 features on Windows.
+// it also doesn't have INTPTR_WIDTH or __INTPTR_WIDTH__ with the preprocessor.
+// to be be fair, Linux also doesn't have INTPTR_WIDTH that early.
+// comparing sizeof(void *) also doesn't work with #if
+
+static inline void *atomic_load(intptr_t volatile *p)
+{
+	// atomic_load uses memory_order_seq_cst, so synchronize here as well
+	intptr_t x;
+	intptr_t comparand = *p;
+	do {
+		x = comparand;
+		if (sizeof(*p) == 8)
+			comparand = _InterlockedCompareExchange64(p, x, comparand);
```

Comment:
Can this (including loop) be replaced with `_InterlockedCompareExchange(p, 0, 0)` ?

P.S. for x86_64 we don't even need to use `atomic_store` and can simply use `volatile`, but to keep thins more reliable, I'm suggesting this.

Reply:
the default ordering for `atomic_` operations is "sequentially consistent" to ensure consistent behaviour when protecting locks (from acquire to release).
it's probably fine to go without but it shouldn't be called `atomic_load` anymore.

imo, the "proper" solution is to add an abstraction that ensures at least "relaxed" ordering.
it would be implemented using `atomic_` or `_Interlocked` and the `atomic_` variant would just be slight overkill.

Reply:
It will be great to avoid overhead here, since then, we may consider using it to resolve #777
Are you willing to make proper primitives or it is better to merge this as-is?